### PR TITLE
feat: add libtiff5 to 22.04

### PR DESCRIPTION
**Pull request by dulmandakh**
# Proposed changes
This PR adds libtiff5 and its dependencies to Ubuntu 22.04

### Forward porting
- 24.04 - https://github.com/canonical/chisel-releases/pull/253

## Testing

```bash
# Example:
#  chisel cut ... --root <path> <pkg>_<proposed-slice>
#  sudo chroot <path> <app>
```

## Checklist

* [x] I have read the [contributing guidelines](
https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
* [x] I have already submitted the [CLA form](
https://ubuntu.com/legal/contributors/agreement)

### CODE REVIEW

**Comment by cjdcordeiro** [comment]
> Diff of dependencies:
> 
> slices/libtiff5.yaml
> ```diff
> @@ -1,7 +1,7 @@
>  libc6
>  libdeflate0
>  libjbig0
> -libjpeg8
> +libjpeg-turbo8
>  liblzma5
>  libwebp7
>  libzstd1
> ```

Thanks for the PR. Can you please comment on this dependency diff?

**Comment by thecoldwine** [comment]
@cjdcordeiro is there any progress or deadline on the PR expected? I do prepare some PRs for 22.04 for libgdiplus which does require libtiff and wonder if I need to open a serial of PRs with slices including this one, since this one seems stale.

**Review by rebornplusplus — COMMENTED** [review]
Looks good to me for the most part. Thank you.

Left a single comment below.

File: slices/libtiff5.yaml
Comment (rebornplusplus) [reply slices/libtiff5.yaml]: Eh, the changes were trivial. I added them here: https://github.com/canonical/chisel-releases/pull/254/commits/6cf65b5f754cbfb9f80b4290b57e03ea393d1744.

**Review by rebornplusplus — APPROVED** [review]
Added the changes I thought this needed. Need another maintainer approval to merge!

**Comment by rebornplusplus** [comment]
> @cjdcordeiro is there any progress or deadline on the PR expected? I do prepare some PRs for 22.04 for libgdiplus which does require libtiff and wonder if I need to open a serial of PRs with slices including this one, since this one seems stale.

Hi Mikhail, sorry about the delay here. I will try to make this go in soon!

**Review by zhijie-yang — APPROVED** [review]
LGTM. Nice trick on libjpeg8.

**Comment by cjdcordeiro** [comment]
> > @cjdcordeiro is there any progress or deadline on the PR expected? I do prepare some PRs for 22.04 for libgdiplus which does require libtiff and wonder if I need to open a serial of PRs with slices including this one, since this one seems stale.
> 
> Hi Mikhail, sorry about the delay here. I will try to make this go in soon!

There one comment that needs to be addressed - FYI @dulmandakh 

But @thecoldwine , if there's no response soon, feel free to open your PR on top of this HEAD

File: slices/libtiff5.yaml
Comment (cjdcordeiro) [reply slices/libtiff5.yaml]: Could you please add a comment explaining it then? Same for #302

**Comment by dulmandakh** [comment]
> > Diff of dependencies:
> 
> > 
> 
> > slices/libtiff5.yaml
> 
> > ```diff
> 
> > @@ -1,7 +1,7 @@
> 
> >  libc6
> 
> >  libdeflate0
> 
> >  libjbig0
> 
> > -libjpeg8
> 
> > +libjpeg-turbo8
> 
> >  liblzma5
> 
> >  libwebp7
> 
> >  libzstd1
> 
> > ```
> 
> 
> 
> Thanks for the PR. Can you please comment on this dependency diff?

I'm trying to imitate Deb dependency. I can change it if unnecessary. Any suggestions?

**Comment by dulmandakh** [comment]
> > > @cjdcordeiro is there any progress or deadline on the PR expected? I do prepare some PRs for 22.04 for libgdiplus which does require libtiff and wonder if I need to open a serial of PRs with slices including this one, since this one seems stale.
> 
> > 
> 
> > Hi Mikhail, sorry about the delay here. I will try to make this go in soon!
> 
> 
> 
> There one comment that needs to be addressed - FYI @dulmandakh 
> 

What was the comment? I'll address it as soon as possible.

**Comment by cjdcordeiro** [comment]
It's related to that dependency diff, but @rebornplusplus already justified it here https://github.com/canonical/chisel-releases/pull/254#discussion_r1700290806

I'd just like to have a comment in the YAML file for future reference, explaining why we use that dependency instead.

**Comment by dulmandakh** [comment]
> It's related to that dependency diff, but @rebornplusplus already justified it here https://github.com/canonical/chisel-releases/pull/254#discussion_r1700290806
> 
> 
> 
> I'd just like to have a comment in the YAML file for future reference, explaining why we use that dependency instead. 

IMHO, there is no comments needed because dependencies are aligned with deb dependencies. We would need if it depended on directly on jpeg turbo package.

**Review by cjdcordeiro — APPROVED** [review]
y all good. merging. thanks

**Comment by thecoldwine** [comment]
Thanks everyone, now I can continue with libgdiplus! :)

**Pull request merged**
